### PR TITLE
Hash ops cleanups

### PR DESCRIPTION
**Pull request by yuwata**
C.f. https://github.com/systemd/systemd/pull/9760#discussion_r206444594.

### CODE REVIEW

**Comment by yuwata** [comment]
@keszybz OK. The most part of the first commit has been dropped. Only exception is the compare function for `NetworkConfigSection`. The element is unsigned, so explicitly comparing the values is safer.

PTAL.

File: src/network/networkd-network.c [hunk 1]
```diff
@@ -36,7 +36,12 @@ static int network_config_compare_func(const void *a, const void *b) {
         if (r != 0)
                 return r;
 
-        return y->line - x->line;
+        if (x->line < y->line)
+                return -1;
+        if (x->line > y->line)
+                return 1;
```
Comment (keszybz) [inline src/network/networkd-network.c:42]: Hmm, that's an interesting bug. So the comparison function was borked all the time and nobody noticed.

**Comment by filbranden** [comment]
How about this?
https://github.com/filbranden/systemd/commit/ca75d503c92c26a4731125f0cc4b872b2d7b249b

(Let me know if you like it, I can polish it and test it, push a PR...)

**Comment by yuwata** [comment]
@filbranden lgtm.

**Pull request merged**